src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -574,7 +575,12 @@ void QgsTemporalControllerWidget::setDates( const QgsDateTimeRange &range )
     whileBlocking( mEndDateTime )->setDateTime( range.end() );
     whileBlocking( mFixedRangeStartDateTime )->setDateTime( range.begin() );
     whileBlocking( mFixedRangeEndDateTime )->setDateTime( range.end() );
-    updateTemporalExtent();
+    // only if range is different from mNavigationObject->temporalExtents update
+    // else recursion as updateTemporalExtent itself calls mNavigationObject->setTemporalExtents
+    if ( range != mNavigationObject->temporalExtents() )
```

Comment:
@rduivenvoorde you will need to add "operator !=" function in the QgsDateTimeRange definition first before using it here. See https://github.com/qgis/QGIS/blob/a51c20136c02b9e5db8f49342f2e6f19a5953476/src/core/qgsrange.h#L453